.changeset/popular-eels-melt.md

```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`CopyButton` - Converted component to TypeScript
```

Comment:
I based this copySnippet branch off the copyButton branch originally. I've tried re-basing onto the newer clipboard branch but not sure how to fix these unrelated changes in here. (Maybe I need to keep it based off of the copyButton branch?)

Reply:
I think you have different options
- just base off the `hds-2688-copy-button-ts-conversion` branch (the simplest)
- rename this branch `hds-2689-copy-snippet-ts-conversion__OLD`, create a new branch with the `hds-2689-copy-snippet-ts-conversion` name and cherry pick only the commits related to the `CopySnippet`, and then force push
- delete from the commit history of this branch the commits that are related to the `hds-clipboard` modifier or the `CopyButton` component, and then force push

let me know if you want we can do it together

Reply:
I decided to keep it simple and just change it back to being based off of `hds-2688-copy-button-ts-conversion`. Thanks!